examples/flash_example.rs

```diff
@@ -93,6 +155,22 @@ fn main() -> ! {
         &mut pac.RESETS,
     );
 
+    // add some delay to give an attached debug probe time to parse the
+    // defmt RTT header. Reading that header might touch flash memory, which
+    // interferes with flash write operations.
+    // https://github.com/knurling-rs/defmt/pull/683
+    delay.delay_ms(10);
+
+    let conf = {
+        let conf = unsafe { CONF_FLASH_SECTOR.read().assume_init() };
```

Comment:
@jannic Thanks a lot for your analysis. It should be possible to accommodate `UnsafeCell` in a similar design, will look into it.

One thing bothers me: you comment about reading uninitialized memory. Being treated literally, it means we cannot **ever** read flash memory before we write it (in the same run, which makes it meaningless), because otherwise we'd risk reading "uninitialized memory", which is an undefined behavior. I believe, you were referring to this piece in `MaybeUninit` documentation:

> Moreover, uninitialized memory is special in that it does not have a fixed value (“fixed” meaning “it won’t change without being written to”). Reading the same uninitialized byte multiple times can give different results. This makes it undefined behavior to have uninitialized data in a variable even if that variable has an integer type, which otherwise can hold any fixed bit pattern

I'm not sure from where this is coming, I'd much prefer to see a reference to some paper or spec or whatever is the basis for this. I'd suspect this may be specific to DRAM, which MCUs do not use (I believe SRAM is used in MCUs due to simpler operation), but the subject case is even more different: we're working with memory-mapped _flash_ memory. We can check the datasheet (or a few), but I'm pretty sure erased flash sectors read _consistently_ as 0xFF (or 0x00, if the chip inverts everything, which I've heard may be a feature of some chips). The fact that uninitialized (erased) flash memory is guaranteed to read consistently eliminates the possibility of undefined behavior due to "unstable" memory as described in the above excerpt.

Thoughts?